hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -414,7 +427,8 @@ namespace hpx { namespace threads { namespace detail
                     // Re-add this work item to our list of work items if the HPX
                     // thread should be re-scheduled. If the HPX thread is suspended
                     // now we just keep it in the map of threads.
-                    if (HPX_UNLIKELY(state_val == pending)) {
+                    if (HPX_UNLIKELY(state_val == pending))
```

Comment:
This is pure gut feeling. I'd suspect more threads to exit in `terminated` and `suspended` state than in `pending` state. But we could certainly verify this. However I don't think these hints have a big impact on the overall performance.